pkg/capacityscheduling/capacity_scheduling.go

```diff
@@ -0,0 +1,869 @@
+	// from the highest priority victims.
+	violatingVictims, nonViolatingVictims := filterPodsWithPDBViolation(potentialVictims, pdbs)
+	reprievePod := func(p *v1.Pod) (bool, error) {
+		if err := addPod(p); err != nil {
+			return false, err
+		}
+		fits, _, _ := core.PodPassesFiltersOnNode(ctx, ph, state, pod, nodeInfo)
+		if !fits {
+			if err := removePod(p); err != nil {
+				return false, err
+			}
+			victims = append(victims, p)
+			klog.V(5).Infof("Pod %v/%v is a potential preemption victim on node %v.", p.Namespace, p.Name, nodeInfo.Node().Name)
+		}
+		return fits, nil
+	}
+	for _, p := range violatingVictims {
+		if fits, err := reprievePod(p); err != nil {
+			klog.Warningf("Failed to reprieve pod %q: %v", p.Name, err)
+			return nil, 0, false
+		} else if !fits {
+			numViolatingVictim++
+		}
+	}
+	// Now we try to reprieve non-violating victims.
+	for _, p := range nonViolatingVictims {
+		if _, err := reprievePod(p); err != nil {
+			klog.Warningf("Failed to reprieve pod %q: %v", p.Name, err)
+			return nil, 0, false
+		}
+	}
+	return victims, numViolatingVictim, true
+}
+
+func (c *CapacityScheduling) addElasticQuota(obj interface{}) {
+	elasticQuota := obj.(*v1alpha1.ElasticQuota)
+	elasticQuotaInfo := &ElasticQuotaInfo{
+		Namespace: elasticQuota.Namespace,
+		Min:       framework.Resource{},
+		Max:       framework.Resource{},
+		Used:      framework.Resource{},
+	}
+	elasticQuotaInfo.Max.Add(elasticQuota.Spec.Max)
+	elasticQuotaInfo.Min.Add(elasticQuota.Spec.Min)
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.elasticQuotas[elasticQuota.Namespace] = elasticQuotaInfo
+}
+
+func (c *CapacityScheduling) updateElasticQuota(oldObj, newObj interface{}) {
+	oldElasticQuota := oldObj.(*v1alpha1.ElasticQuota)
+	newElasticQuota := newObj.(*v1alpha1.ElasticQuota)
+	newElasticQuotaInfo := &ElasticQuotaInfo{
+		Namespace: newElasticQuota.Namespace,
+		Min:       framework.Resource{},
+		Used:      framework.Resource{},
+	}
+	newElasticQuotaInfo.Max.Add(newElasticQuota.Spec.Max)
+	newElasticQuotaInfo.Min.Add(newElasticQuota.Spec.Min)
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	oldElasticQuotaInfo, exit := c.elasticQuotas[oldElasticQuota.Namespace]
+	if exit {
+		newElasticQuotaInfo.Used = oldElasticQuotaInfo.Used
+	}
+	c.elasticQuotas[newElasticQuota.Namespace] = newElasticQuotaInfo
+}
+
+func (c *CapacityScheduling) deleteElasticQuota(obj interface{}) {
+	elasticQuota := obj.(*v1alpha1.ElasticQuota)
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	delete(c.elasticQuotas, elasticQuota.Namespace)
+}
+
+func (c *CapacityScheduling) addPod(obj interface{}) {
+	pod := obj.(*v1.Pod)
+	isAssumed, err := c.isAssumedPod(pod)
+	if err != nil {
+		klog.Errorf("failed to check whether pod %s/%s is assumed: %v", pod.Namespace, pod.Name, err)
+		return
+	}
+	if isAssumed {
+		klog.V(4).Infof("pod %s/%s is assumed", pod.Namespace, pod.Name)
+		c.deleteAssumedPod(pod)
+		return
+	}
+	elasticQuotaInfo, exit := c.elasticQuotas[pod.Namespace]
+	if exit {
```

Comment:
https://github.com/kubernetes-sigs/scheduler-plugins/pull/73/files#diff-d2629e9bc4a82622042892618ce271fef08fca9723f7d2993f16e9099db78134R162-R164

The plugins will sync the elastic event firstly. I think this will solve the problem of sequence of events like the scheduler restart?